modules/scan/soon_to_expire_certificate.yaml

```diff
@@ -0,0 +1,48 @@
+info:
+  id: soon_to_expire_certificate_scan
+  author: OWASP Nettacker Team
+  severity: 3
+  description: Find certificates which will be expire less than 30days.
+  reference:
+  profiles:
+    - scan
+    - information_gathering
+    - infortmation
+    - info
+    - low_severity
+    - ssl
+    - tls
+
+payloads:
+  - library: exec
+    steps:
+      - method: os_command
+        command:
+          nettacker_fuzzer:
+            input_format: "echo | openssl s_client -servername {target} -connect {target}:{{ports}} 2> /dev/null | openssl x509 -noout -enddate | cut -d= -f2"
+            prefix: ""
+            suffix: ""
+            interceptors:
+            data:
+              ports:
+                - 80 # HTTP (some LBs support TLS on 80)
+                - 261 # Nsiiops
+                - 443 # HTTPS
+                - 446 # Openfiler's management interface
+                - 448 # ddm-ssl
+                - 465 # SMTPS
+                - 563 # NNTPS
+                - 585 # imap4-ssl
+                - 614 # SSLshell
+                - 636 # LDAPS
+                - 684 # Corba IIOP SSL
+                - 695 # IEEE-MMS-SSL
+                - 853 # DNS over TLS
+                - 902 # VMWare Auth Daemon
+                - 989 # FTPS data
+        response:
+          condition_type: or
+          conditions:
+            END_DATE:
+              regex: < 2592000 # seconds (30days)
```

Comment:
it will grab expired one too, naturally endate < 2592000 and also 0 < 2592000, expired one will be gathered as in "untrusted_certificate_scan" too.